src/connection.js

```diff
@@ -1245,6 +1286,7 @@ Connection.prototype.STATE = {
     name: 'SentClientRequest',
     exit: function() {
       this.clearRequestTimer();
+      this.tokenStreamParser.resume();
```

Comment:
Request class has 'paused' state now and that'll be inconsistent with the actual state of the request at this point.
Move the 'paused' state to Connection class and have Request pause/resume methods just forward the calls to Connection. That'll remove this coupling.